packages/backend-function/src/factory.ts

```diff
@@ -318,6 +318,23 @@ class FunctionFactory implements ConstructFactory<AmplifyFunction> {
     return this.props.memoryMB;
   };
 
+  private resolveEnvironment = () => {
+    if (this.props.environment === undefined) {
+      return {};
+    }
+
+    Object.keys(this.props.environment).forEach((key) => {
+      // validate using key pattern from https://docs.aws.amazon.com/lambda/latest/api/API_Environment.html
+      if (!key.match(/^[a-zA-Z]([a-zA-Z0-9_])+$/)) {
+        throw new Error(
```

Comment:
Shouldn't this be `AmplifyUserError`?

Reply:
After comparing with other packages I believe you're right, though this error and others in `FunctionFactory` are caught by https://github.com/aws-amplify/amplify-backend/blob/main/packages/backend-deployer/src/cdk_error_mapper.ts#L352. I'll update all of them to `AmplifyUserError`.